kedro/framework/cli/micropkg.py

```diff
@@ -267,7 +280,219 @@ def package_pipeline(
+
+    protocol, _ = get_protocol_and_path(location)
+    fs_args_config = anyconfig.load(fs_args) if fs_args else {}
+
+    try:
+        return fsspec.filesystem(protocol, **fs_args_config)
+    except Exception as exc:  # pylint: disable=broad-except
+        # Specified protocol is not supported by `fsspec`
+        # or requires extra dependencies
+        click.secho(str(exc), fg="red")
+        click.secho("Trying to use 'pip download'...", fg="red")
+        return None
+
+
+def _unpack_sdist(location: str, destination: Path, fs_args: Optional[str]) -> None:
+    filesystem = _get_fsspec_filesystem(location, fs_args)
+
+    if location.endswith(".tar.gz") and filesystem and filesystem.exists(location):
+        with filesystem.open(location) as fs_file:
+            with tarfile.open(fileobj=fs_file, mode="r:gz") as tar_file:
+                tar_file.extractall(destination)
+    else:
+        python_call(
+            "pip", ["download", "--no-deps", "--dest", str(destination), location]
+        )
+        sdist_file = list(destination.glob("*.tar.gz"))
+        # `--no-deps` should fetch only one source distribution file, and CLI should fail if that's
+        # not the case.
+        if len(sdist_file) != 1:
+            file_names = [sf.name for sf in sdist_file]
+            raise KedroCliError(
+                f"More than 1 or no sdist files found: {file_names}. "
+                f"There has to be exactly one source distribution file."
+            )
+        with tarfile.open(sdist_file[0], "r:gz") as fs_file:
+            fs_file.extractall(destination)
+
+
+def _rename_files(conf_source: Path, old_name: str, new_name: str):
+    config_files_to_rename = (
+        each
+        for each in conf_source.rglob("*")
+        if each.is_file() and old_name in each.name
+    )
+    for config_file in config_files_to_rename:
+        new_config_name = config_file.name.replace(old_name, new_name)
+        config_file.rename(config_file.parent / new_config_name)
+
+
+def _refactor_code_for_unpacking(
+    project: Project,
+    package_path: Path,
+    tests_path: Path,
+    alias: Optional[str],
+    destination: Optional[str],
+    project_metadata: ProjectMetadata,
+) -> Tuple[Path, Path]:
+    """This is the reverse operation of `_refactor_code_for_package`, i.e
+    we go from:
+    <temp_dir>  # also the root of the Rope project
+    |__ <pipeline_name>  # or <alias>
+        |__ __init__.py
+    |__ tests  # only tests for <pipeline_name>
+        |__ __init__.py
+        |__ test_pipeline.py
+
+    to:
+    <temp_dir>  # also the root of the Rope project
+    |__ <package>
+        |__ __init__.py
+        |__ <path_to_pipeline>
+            |__ __init__.py
+            |__ <pipeline_name>
+                |__ __init__.py
+    |__ tests
+        |__ __init__.py
+        |__ <path_to_pipeline>
+            |__ __init__.py
+            |__ <pipeline_name>
+                |__ __init__.py
```

Comment:
Might be nice to update these docstrings too, to refer to `project_package` and `micro-package` etc.